### DISCUSSION TITLE
How to use https mode in gRPC-Web

### DISCUSSION THREAD

**Discussion by dlyltm** [grpc/grpc-web#1434] [Q&A - General]
In "gRPC-Web Hello World" example, there is "var client = new GreeterClient('http://localhost:8080');" in client code. http is used in it. Now I want to change http to https and ignore invalid ssl certificates, but I've tried many times and failed. What should I do about this?

**Comment by sampajano** [comment]
@dlyltm Hi :)

If you can provide more details about the failure, it might help understand the issue.

Your question do sound like a generic issue where Google or Chatgpt could help with :)

**Reply by dlyltm** [reply]
When I changed "var client = new GreeterClient('http://localhost:8080/');" to "var client = new GreeterClient('https://_ip:port_');",the following error message is displayed in the browser:
![Snipaste_2024-06-13_14-19-07](https://github.com/grpc/grpc-web/assets/33916145/24322f1e-7fb2-4e92-aa74-6ccf3ab7f0e3)
I also used AI to make many queries, but the answers didn't work.

**Reply by sampajano** [reply]
Aha :)

I tried asking ChatGPT 4o and here's the beginning of the answer. It seems to be a pretty involved process so unless you did everything right then it's quite expected that this won't work :)

<img width="633" alt="Screenshot 2024-06-13 at 10 25 35 AM" src="https://github.com/grpc/grpc-web/assets/133680/c0696864-6a59-48ad-8a75-a943530e9d76">

(... more pages omitted)

---

I asked it to summarize and here's the answer i got :)

<img width="644" alt="Screenshot 2024-06-13 at 10 27 29 AM" src="https://github.com/grpc/grpc-web/assets/133680/dc4da375-8b30-417e-b95f-4ca1a2d3227b">

Hope this helps :)

**Reply by dlyltm** [reply]
Thanks for your answer. My original intention is to ignore invalid ssl certificates but not to create a self-signed certificate. I did it in the .net core client and it worked, but it doesn't work in the browser, ut supra. I just want to know how to write this code in js.

**Reply by sampajano** [reply]
Aha ok! You're welcome! Glad it's working for you now :)
